src/wasm/wasm-type.cpp

```diff
@@ -1241,6 +1252,35 @@ HeapType HeapType::getLeastUpperBound(HeapType a, HeapType b) {
   return TypeBounder().getLeastUpperBound(a, b);
 }
 
+RecGroup HeapType::getRecGroup() const {
+  assert(!isBasic());
+  if (auto* info = getHeapTypeInfo(*this)->recGroup) {
+    return RecGroup(uintptr_t(info));
+  } else {
+    // Mark the low bit to signify that this is a trivial recursion group and
+    // points to a heap type info rather than a vector of heap types.
+    return RecGroup(id | 1);
+  }
+}
+
+std::vector<HeapType> RecGroup::getHeapTypes() const {
+  if (id & 1) {
+    // This is a trivial recursion group. Mask off the low bit to recover the
+    // single HeapType.
```

Comment:
Where is this encoding defined/documented? Sorry if I missed it.

Reply:
It's not, really 😅 Will add documentation.